### PR TITLE
NOISSUE - Refactor InfluxDB Reader: explicit check event + add safe conversion

### PR DESCRIPTION
**Pull request by tzzed**
### What does this do?
This PR is a simple refactoring of influxdb measurement data.

### Which issue(s) does this PR fix/relate to?
No issue related to this PR.

### Have you included tests for your changes?
Still working
### Did you document any new/modified functionality?
No

### CODE REVIEW

File: readers/influxdb/messages.go [hunk 1]
```diff
@@ -239,8 +239,13 @@ func parseSenml(names []string, fields []interface{}) interface{} {
 				msgField.SetString(s)
 			}
 		case float64:
+			value, ok := fields[i].(string)
```
Comment (drasko) [inline readers/influxdb/messages.go]: `value` is not the best name for this variable. Maybe `f`, or `fs` (like field-string`) or something else would be more go-idiomatic

File: readers/influxdb/messages_test.go [hunk 2]
```diff
@@ -498,7 +498,7 @@ func TestReadJSON(t *testing.T) {
 
 	for desc, tc := range cases {
 		result, err := reader.ReadAll(tc.chanID, tc.pageMeta)
-
+
```
Comment (drasko) [inline readers/influxdb/messages_test.go]: Please remove this blank line

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
@tzzed thanks for the PR. Please sync your branch to the master.

@arvindh123 please review this PR

File: readers/influxdb/messages_test.go (hunk 2 above)
Comment (tzzed) [reply readers/influxdb/messages_test.go]: Error handling is missing here,
should be is this wanted or should ´require.NoError´

File: readers/influxdb/messages.go [hunk 3]
```diff
@@ -7,10 +7,10 @@ import (
 	"strconv"
 	"strings"
 	"time"
-
+
```
Comment (manuio) [inline readers/influxdb/messages.go]: Looks like you are using spaces to complete the tabulation. Can you please remove it here and in all other places.

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by arvindh123 — COMMENTED** [review]
@drasko, It looks good

File: readers/influxdb/messages_test.go [hunk 4]
```diff
@@ -498,15 +498,15 @@ func TestReadJSON(t *testing.T) {
...
+		require.NoError(t, err)
+		
 		for i := 0; i < len(result.Messages); i++ {
 			m := result.Messages[i]
 			// Remove time as it is not sent by the client.
 			delete(m.(map[string]interface{}), "time")
 
 			result.Messages[i] = m
 		}
-		assert.Nil(t, err, fmt.Sprintf("%s: expected no error got %s", desc, err))
```
Comment (drasko) [inline readers/influxdb/messages_test.go:509]: Why is this assert removed?

**Review by drasko — CHANGES_REQUESTED** [review]
Honestly, I do not see value in this PR. I would like that @dusanb94 reviews, but I do not find new code better or more comprehensible.

File: readers/influxdb/messages_test.go (hunk 4 above)
Comment (tzzed) [reply readers/influxdb/messages_test.go:509]: The assert checks the error after the loop but it should be done before just after getting the `result, err := reader.ReadAll(tc.chanID, tc.pageMeta)` .

**Comment by tzzed** [comment]
the len's comparison to zero is more clear because we want to check the 0 value an empty array. `len(v)  < 1` includes negatives values.

About the safe cast is just to avoid panic. 

I just let myself to do this PR as a "good first issue" but you are free to close it.

**Comment by drasko** [comment]
@nmarcetic and @dusanb94 need to review and approve this one.

File: readers/influxdb/messages.go [hunk 5]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/mainflux/mainflux/readers"
 
 	influxdata "github.com/influxdata/influxdb/client/v2"
+
```
Comment (drasko) [inline readers/influxdb/messages.go]: Why this blank line?

File: readers/influxdb/messages.go [hunk 6]
```diff
@@ -53,28 +54,27 @@ func (repo *influxRepository) ReadAll(chanID string, rpm readers.PageMetadata) (
...
 		return readers.MessagesPage{}, errors.Wrap(errReadMessages, resp.Error())
 	}
 
-	if len(resp.Results) < 1 || len(resp.Results[0].Series) < 1 {
+	if len(resp.Results) == 0 || len(resp.Results[0].Series) == 0 {
 		return readers.MessagesPage{}, nil
 	}
 
+	var messages []readers.Message
+
```
Comment (drasko) [inline readers/influxdb/messages.go]: No need for blank line

File: readers/influxdb/messages.go [hunk 7]
```diff
@@ -140,7 +140,8 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
 	if err != nil {
 		return condition
 	}
-	json.Unmarshal(meta, &query)
+
```
Comment (drasko) [inline readers/influxdb/messages.go:141]: No need for blank line.

File: readers/influxdb/messages.go [hunk 8]
```diff
@@ -140,7 +140,8 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
...
+	_ = json.Unmarshal(meta, &query)
```
Comment (drasko) [inline readers/influxdb/messages.go]: Do not ignore errors.

File: readers/influxdb/messages.go [hunk 9]
```diff
@@ -187,28 +188,30 @@ func parseValues(value interface{}, name string, msg *senml.Message) {
...
-				return
-			}
-			msg.Value = &num
-		case string:
-			if strings.HasPrefix(name, "string") {
-				v := value.(string)
-				msg.StringValue = &v
-				return
-			}
+	if !strings.HasSuffix(strings.ToLower(name), "value") {
```
Comment (drasko) [inline readers/influxdb/messages.go]: I find this harder to read than it was before.

File: readers/influxdb/messages_test.go (hunk 2 above)
Comment (drasko) [reply readers/influxdb/messages_test.go]: I meant the empty (blank) line that you added below.

File: readers/influxdb/messages_test.go (hunk 4 above)
Comment (drasko) [reply readers/influxdb/messages_test.go:509]: Is it done?

**Review by drasko — CHANGES_REQUESTED** [review]

File: readers/influxdb/messages.go (hunk 5 above)
Comment (tzzed) [reply readers/influxdb/messages.go]: added by goimport. I remove it.

**Comment by dborovcanin** [comment]
> Honestly, I do not see value in this PR. I would like that @dusanb94 reviews, but I do not find new code better or more comprehensible.

The refactoring makes sense. @tzzed please consider adding better error handling and InfluxDB records parsing. Those improvements can be added to this PR to improve its value.

File: readers/influxdb/messages.go (hunk 5 above)
Comment (drasko) [reply readers/influxdb/messages.go]: Please remove this blank line.

File: readers/influxdb/messages_test.go (hunk 4 above)
Comment (drasko) [reply readers/influxdb/messages_test.go:509]: Again - is this done?

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
> > Honestly, I do not see value in this PR. I would like that @dusanb94 reviews, but I do not find new code better or more comprehensible.
> 
> The refactoring makes sense. @tzzed please consider adding better error handling and InfluxDB records parsing. Those improvements can be added to this PR to improve its value.

OK, then I left a few comments, and these must be corrected before we are ready to merge.

**Comment by drasko** [comment]
@tzzed if there is no action from your side to correct mentioned comment, this PR will be closed due to the inactivity.

File: readers/influxdb/messages_test.go (hunk 4 above)
Comment (tzzed) [reply readers/influxdb/messages_test.go:509]: I put it  after `reader.ReadAll(tc.chanID, tc.pageMeta)`

File: readers/influxdb/messages.go (hunk 5 above)
Comment (tzzed) [reply readers/influxdb/messages.go]: It is set by goimport.

Comment (drasko) [reply readers/influxdb/messages.go]: No, it is not. Goimport does not add blank lines. Maybe your editor does. Just delete this line and please do not resolve remarks by hand yourself. Reviewers will resolve your remarks if they are handled.

File: readers/influxdb/messages_test.go (hunk 4 above)
Comment (manuio) [reply readers/influxdb/messages_test.go:509]: This looks ok but in any case don't resolve remarks by hand please.

File: readers/influxdb/messages_test.go [hunk 10]
```diff
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	influxdata "github.com/influxdata/influxdb/client/v2"
+
```
Comment (manuio) [inline readers/influxdb/messages_test.go]: Remove this blank line

Comment (tzzed) [reply readers/influxdb/messages_test.go]: done

**Comment by drasko** [comment]
@tzzed CI is failing. Please check your tests, you have introduced regression:

```
    messages_test.go:378: 
        	Error Trace:	messages_test.go:378
        	Error:      	Not equal: 
        	            	expected: 0x4f
        	            	actual  : 0x64
        	Test:       	TestReadAll
        	Messages:   	read message with to: expected 79 got 100
    messages_test.go:378: 
        	Error Trace:	messages_test.go:378
        	Error:      	Not equal: 
        	            	expected: 0x15
        	            	actual  : 0x64
        	Test:       	TestReadAll
        	Messages:   	read message with from: expected 21 got 100
--- FAIL: TestReadAll (0.62s)
=== RUN   TestReadJSON
--- PASS: TestReadJSON (0.12s)
FAIL
coverage: 86.4% of statements
FAIL	github.com/mainflux/mainflux/readers/influxdb	20.089s
FAIL
```

**Comment by tzzed** [comment]
@drasko I did not introduced regression. This is because of the `msg.Time` setting.  `now` is in second but when it is equal to 0, 5, or 20 it fails. I invite you to try to set it into this 3 values in the master branch and run the test.

```go
now := float64(time.Now().UTC().Second())

	for i := 0; i < msgsNum; i++ {
		// Mix possible values as well as value sum.
		msg := m
		msg.Time = now - float64(i)  // This operations has a random fails or success.
```

I will change it tomorrow if you are ok @drasko ? It will add a real value to this PR.

File: readers/influxdb/messages_test.go [hunk 11]
```diff
@@ -71,7 +71,10 @@ func TestReadAll(t *testing.T) {
 	stringMsgs := []senml.Message{}
 	dataMsgs := []senml.Message{}
 	queryMsgs := []senml.Message{}
-	now := float64(time.Now().UTC().Second())
+	rand.Seed(time.Now().UnixNano())
+	to := msgsNum
+	from := 21
```
Comment (dborovcanin) [inline readers/influxdb/messages_test.go]: You can use this as a constant in tests with a range search instead of literals, as well.

Comment (dborovcanin) [reply readers/influxdb/messages_test.go]: @tzzed I also meant to use this constant later in the code: https://github.com/mainflux/mainflux/blob/master/readers/influxdb/messages_test.go#L334-L355

Comment (tzzed) [reply readers/influxdb/messages_test.go]: I have seen this line but I have not enough context about this lines. How do you see the usage.Should I create a const for each value in test. 
```go
pageMeta: readers.PageMetadata{
				Offset: 0,
				Limit:  uint64(len(messages[21:])),
				To:     messages[20].Time,
			},
```

Comment (dborovcanin) [reply readers/influxdb/messages_test.go]: No need, I think. Replacing literals `20` and `21` with `from` and `from + 1`  (and maybe rename `from` to `offset`) will be just fine.

**Review by drasko — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Review by blokovi — APPROVED** [review]

**Pull request merged**